tests/test_agent_manager.py

```diff
@@ -1272,3 +1273,25 @@ async def test_dont_start_paused_agent(server, client, environment, caplog) -> N
     assert "Started new agent with PID" not in caplog.text
     # Ensure no timeout happened
     assert "took too long to start" not in caplog.text
+
+
+async def test_auto_started_agent_log_in_debug_mode(server, environment, monkeypatch):
+    """
+    Test the logging of an autostarted agent
+    """
+    env = await data.Environment.get_by_id(uuid.UUID(environment))
+    await env.set(data.AUTOSTART_AGENT_MAP, {"internal": "", "test1": ""})
+
+    agentmanager = server.get_slice(SLICE_AGENT_MANAGER)
+    autostarted_agentmanager = server.get_slice(SLICE_AUTOSTARTED_AGENT_MANAGER)
+
+    await agentmanager.ensure_agent_registered(env, "test1")
+    await autostarted_agentmanager._ensure_agents(env, ["test1"])
+
+    logdir = Config.get("config", "log-dir")
+    log_file = f"{logdir}/agent-{environment}.log"  # Path to the log file
+
+    with open(log_file, mode="r") as f:
+        log_content = f.read()
+
+    assert "DEBUG    inmanta.protocol.endpoints Start transport for client agent" in log_content
```

Comment:
There is a race condition here. If the agent starts slowly, the log entry might not be there yet. We should use `retry_limited` here.

Reply:
indeed, I didn't think of that. I'll add it. Thanks!